src/protocol/codecs/topic.rs

```diff
@@ -28,9 +30,17 @@ impl MsgOfInterest {
             return Err(invalid_data!("expected 'tags' topic"));
         }
 
-        Ok(Self {
-            tags: tag_topic.value,
-        })
+        let tags = tag_topic.value.split(',');
+        let expected_len = tags.clone().count();
+        let tags = tags
+            .map_while(|tag| Tag::try_from(tag).ok())
+            .collect::<HashSet<Tag>>();
+
+        if tags.len() != expected_len {
+            return Err(invalid_data!("invalid tag received"));
+        }
```

Comment:
this is fine, but I'd instead go with the following:
```
let tags = tags
            .map(Tag::try_from)
            .collect::<io::Result<HashSet<Tag>>>()?;
```
there is no need to check the number of commas then, I think

Reply:
alternatively `let tags: HashSet<_> = (...)` (without the explicit type by `collect`), which should work too, but is shorter

Reply:
Great! I was trying to find a solution like that but couldn't make it work... 😅
Thanks!

Reply:
Actually, collect still expects some type annotations:

This is a compromise:
```rust
        let tags: HashSet<_> = tag_topic
            .value
            .split(',')
            .map(Tag::try_from)
            .collect::<io::Result<_>>()?;
```

Reply:
Done